Calculator/Form1.cs

```diff
@@ -295,6 +295,133 @@ private void PerformOperation()
                         textbox_result.Text = resultValue.ToString();
                         break;
                     }
+                case Operation.RaiseNumberToPower:
+                    {
+                        resultValue = Math.Pow(firstTerm, secondTerm);
+                        textbox_result.Text = resultValue.ToString();
+                        break;
+                    }
+                case Operation.SquareRoot:
+                    {
+                        resultValue = Math.Sqrt(firstTerm);
+                        textbox_result.Text = resultValue.ToString();
+                        break;
+                    }
+                case Operation.NumberParity:
+                    {
+                        if (firstTerm % 2 == 0)
+                        {
+                            lbl_title.Text = "The number is even.";
+                        }
+                        else
+                        {
+                            lbl_title.Text = "The number is odd.";
+                        }
+                        break;
+                    }
+                case Operation.MirroredNumber:
+                    {
+                        uint resultValue = 0;
+                        uint number = Convert.ToUInt32(firstTerm);
+                        while (number != 0)
+                        {
+                            resultValue = resultValue * 10 + number % 10;
+                            number /= 10;
+                        }
+                        lbl_title.Text = $"The mirrored number is {resultValue}";
+                        break;
+                    }
+                case Operation.IsPrime:
+                    {
+                        uint number = Convert.ToUInt32(firstTerm);
+                        if (number == 0 || number == 1)
+                        {
+                            lbl_title.Text = $"{number} is not a prime number.";
+                        }
+                        for (uint divisor = 2; divisor <= Math.Sqrt(firstTerm); divisor++)
+                        {
+                            if (number % divisor == 0)
```

Comment:
this is actually a bug. by this algorithm, 18 will appear to be a prime number, which is false.
Please think about what's wrong and fix